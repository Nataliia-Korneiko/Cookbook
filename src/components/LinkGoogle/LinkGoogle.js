import React from 'react';
import { ReactComponent as IconGoogle } from '../../assets/icons/icon-google.svg';
import s from './LinkGoogle.module.css';

const LinkGoogle = () => {
  return (
    <a className={s.link__google} href="https://www.google.com.ua/">
      <div className={s.link__wrapper}>
        <IconGoogle width="28" height="28" />
        <span className={s.link__span}>Google</span>
      </div>
    </a>
  );
};

export default LinkGoogle;
