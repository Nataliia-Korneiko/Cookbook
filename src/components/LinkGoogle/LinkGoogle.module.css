.link__google {
  display: block;
  width: 120px;
  height: 44px;
  border: 2px solid var(--color-golden);
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  font-weight: 500;
  color: var(--color-black);
  background-color: var(--bg-color-white);
  box-shadow: var(--box-shadow-light);
  outline: none;
  transition: 0.8s ease-in-out;
  box-shadow: var(--box-shadow-light);
}

.link__google:hover,
.link__google:focus {
  background-color: var(--bg-color-golden);
  transition: 0.8s ease-in-out;
}

.link__wrapper {
  display: flex;
  padding-left: 8px;
  padding-top: 6px;
}

.link__span {
  padding-left: 6px;
  line-height: 28px;
}
