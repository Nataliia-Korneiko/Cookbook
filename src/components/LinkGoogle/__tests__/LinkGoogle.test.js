/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import LinkGoogle from '../LinkGoogle';

describe('LinkGoogle component', () => {
  const newLinkGoogle = shallow(<LinkGoogle />);

  it('renders a properly', () => {
    expect(newLinkGoogle).toMatchSnapshot();
  });

  it('renders one a', () => {
    expect(
      newLinkGoogle.find('a[href="https://www.google.com.ua/"]'),
    ).toHaveLength(1);
  });

  it('renders one div', () => {
    expect(newLinkGoogle.find('div')).toHaveLength(1);
  });

  it('renders one span', () => {
    expect(newLinkGoogle.find('span')).toHaveLength(1);
  });
});
