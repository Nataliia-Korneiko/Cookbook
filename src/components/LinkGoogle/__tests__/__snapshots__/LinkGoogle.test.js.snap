// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkGoogle component renders a properly 1`] = `ShallowWrapper {}`;
